src/Features/CSharp/Portable/ExtractMethod/CSharpSelectionResult.ExpressionResult.cs

```diff
@@ -44,9 +45,22 @@ public override bool ContainingScopeHasAsyncKeyword()
                 var firstToken = GetFirstTokenInSelection();
                 var lastToken = GetLastTokenInSelection();
                 var scope = firstToken.GetCommonRoot(lastToken).GetAncestorOrThis<ExpressionSyntax>();
-                return scope.IsRightSideOfDotOrArrowOrColonColon()
```

Comment:
during my investigation i found that the `IsRight` and `IsName` members are really really confusing in terms of what they do and they go against the spirit and designs of ISyntaxFacts.  i.e. `IsNameOfMemberAccessExpression` should only be true if something really is the name of a MemberAccessExpressionSyntax (for either language).  It should not also be true if something is the name of a MemberBindingExpression (which is not a MemberAccessExpression).  

This has led to a lot of confusion (and potentially bogus code in many places), so i'm using this as an opportuity to clean this up and make thigns more explicit.